conda_forge_tick/migrators/migration_yaml.py

```diff
@@ -127,3 +133,117 @@ def order(
         return sorted(
             graph, key=lambda x: (len(nx.descendants(total_graph, x)), x), reverse=True,
         )
+
+
+class MigrationYamlCreator(Migrator):
+    """Migrator creating migration yaml files."""
+
+    migrator_version = 0
+    rerender = False
+
+    # TODO: add a description kwarg for the status page at some point.
+    # TODO: make yaml_contents an arg?
+    def __init__(
+        self,
+        package_name: str,
+        new_pin_version: str,
+        current_pin: str,
+        pin_spec: str,
+        pr_limit: int = 1,
+        bump_number: int = 1,
+        **kwargs: Any,
+    ):
+        super().__init__(pr_limit=pr_limit,)
+        self.pin_spec = pin_spec
+        self.current_pin = current_pin
+        self.new_pin_version = new_pin_version
+        self.package_name = package_name
+        self.bump_number = bump_number
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        if attrs["name"] == "conda-forge-pinning":
+            return False
+        return True
+
+    def migrate(
+        self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any
+    ) -> "MigrationUidTypedDict":
+        migration_yaml_dict = {
+            "__migrator": {"build_number": 1, "kind": "version", "migration_number": 1},
+            self.package_name: [self.new_pin_version],
+            "migrator_ts": f"{time.time():.0f}",
+        }
+        with indir(os.path.join(recipe_dir, "migrations")):
+            with open(f"{self.package_name}{self.new_pin_version}.yaml", "w") as f:
+                yaml.dump(migration_yaml_dict, f)
+            eval_xonsh("git add .")
+
+        with indir(recipe_dir):
+            self.set_build_number("meta.yaml")
+
+        with indir(recipe_dir):
+            with open("meta.yaml", "r") as fp:
+                new_lines = []
+                found_ver = False
+                for line in fp.readlines():
+                    if line.startswith("{% set version = "):
+                        now = datetime.datetime.now().strftime("%Y.%m.%d")
```

Comment:
what if we make two PRs in the same day?

Reply:
Then the people who merge those PRs work that out, just like they do with the normal migration bot.

Reply:
Once the first one is merged, the second will get a conflict due to the build number bump.